### PR TITLE
Enhance multiplier/imbue fix

### PR DESCRIPTION
**Pull request by vuzaldo**
Fix for #88 which affects multiple challenge bges where the related skill can be further enhanced by a hero imbue after the 50% increase.

Previous code resulting in `-mult + x` has been changed to stack each type separately and resolve the enhanced amount as `Math.ceil(base * mult) + x`

The debug output for imbue currently works as "imbues existing skill X by X" where X is the same imbued amount, so also added an optional debug function to find the existing target skill and show its current (enhanced) value instead.

### CODE REVIEW

File: dist/practice.js [hunk 1]
```diff
@@ -1052,10 +1052,9 @@ var makeUnit = (function() {
 
 var getEnhancement = function(card, s, base) {
     var enhancements = card.enhanced;
-    var enhanced = (enhancements ? (enhancements[s] || 0) : 0);
-    if (enhanced < 0) {
-        enhanced = Math.ceil(base * -enhanced);
-    }
+    var e = { 'x': 0, 'mult': 0 }; // Default value
```
Comment (TheSench) [inline dist/practice.js]: **nit:** Prefer non-quoted property keys for consistency with the rest of the codebase.

Ditto throughout.

File: dist/practice.js [hunk 2]
```diff
@@ -3204,7 +3220,7 @@ var SIM_CONTROLLER = (function () {
 				// Check Nullify
 				if (target.nullified && !skill.ignore_nullify) {
 					target.nullified--;
-					if (debug) echo += debug_name(src_card) + ' wing guards ' + debug_name(target) + ' but it is nullified!\n';
+					if (debug) echo += debug_name(src_card) + ' wing guards ' + debug_name(target) + ' but it is nullified!<br>';
```
Comment (TheSench) [inline dist/practice.js:3222]: 💚 Thanks for cleaning up this log message.  Good catch on the incorrect use of `\n` over `<br>`.

Normally I'd suggest putting this in a separate PR, as it's unrelated to the main purpose of this one, but it's smaller enough to not require its own PR.

File: dist/practice.js [hunk 3]
```diff
@@ -4295,16 +4311,17 @@ var SIM_CONTROLLER = (function () {
 				// Check Nullify
 				if (target.nullified && !skill.ignore_nullify) {
 					target.nullified--;
-					if (debug) echo += debug_name(src_card) + ' enhances ' + debug_name(target) + ' but it is nullified!<br>';
+					if (debug) echo += debug_name(src_card) + ' imbues ' + debug_name(target) + ' but it is nullified!<br>';
```
Comment (TheSench) [inline dist/practice.js:4313]: 💚 Another good catch.

There's enough logging fixes here to be their own PR at this point, but I'm fine with leaving them in this one, now that they're here.

File: dist/simulator.js [hunk 4]
```diff
@@ -1570,6 +1569,23 @@ function debug_name(card, hideStats) {
     return output;
 }
 
+function debug_find_skill(target, s) {
```
Comment (TheSench) [inline dist/simulator.js:1572]: 💡 How about we update `debug_skill` to take in `target` and then include enhancement when it calculates `x`.

That would keep the logs for `imbue` and `enhance` consistent.

**Review by TheSench — APPROVED** [review]

File: dist/practice.js (hunk 2 above)
Comment (vuzaldo) [reply dist/practice.js:3222]: I pushed the incorrect newline in one of the first PRs and didn't notice it (oops).
Got it about using separate PRs.

**Comment by vuzaldo** [comment]
Amazing changes!
It's definitely more consistent now with `enhance` showing complete information as well.
Also realized a bit too late that `hasSkill` could have been used for better debugging the existing skill.
Now `debug_find_skill` wouldn't need `copy_skill` anymore since it doesn't modify `.x`
